karma.conf.js

```diff
@@ -34,16 +34,21 @@ module.exports = (config) => {
   if (useCloud) {
     Object.assign(config, {
       browsers: Object.keys(browsers),
-      browserDisconnectTolerance: 3,
       // My current OS plan allows max 2 parallel connections.
       concurrency: 2,
-      retryLimit: 3
+      retryLimit: 3,
+
+      // Timeouts taken from https://github.com/karma-runner/karma-browserstack-launcher/issues/61
+      captureTimeout: 3e5,
+      browserNoActivityTimeout: 3e5,
+      browserDisconnectTimeout: 3e5,
+      browserDisconnectTolerance: 3
     })
 
     config.browserStack = {
       username: browserStackUserName,
       accessKey: browserStackAccessKey,
-      captureTimeout: 10000
+      captureTimeout: 3e5
```

Comment:
See this issue: https://github.com/karma-runner/karma-browserstack-launcher/issues/61.
I had issues with testing on the mobile android device which got restarted on every run and then get stuck forever. I had to kill the worker on each run manually in order to free the slot again. I took the values from the issue above, which are supposed to be timeout values given by the browserstack support.